BlueprintUI/Sources/Layout/ConstrainedSize.swift

```diff
@@ -96,6 +96,26 @@ public extension Element {
     {
         ConstrainedSize(width: width, height: height, wrapping: self)
     }
+    
+    /// Constrains the measured size of the element to the provided `SizeConstraint`.
+    func constrained(to sizeConstraint : SizeConstraint) -> ConstrainedSize
+    {
+        ConstrainedSize(
+            width: {
+                switch sizeConstraint.width {
+                case .atMost(let value): return .atMost(value)
+                case .unconstrained: return .unconstrained
+                }
+            }(),
```

Comment:
nit: consider an inline func for this instead of the closures

Reply:
I've always preferred these sort of closures since it's less characters and nests the relevant value provider at the callsite; though if there was more than two parameters here; I'd agree